### DISCUSSION TITLE
Using jiten database

### DISCUSSION THREAD

**Discussion by luckasRanarison** [obfusk/jiten#154] [Q&A] [answered]
Hi there! Thanks for this amazing project, I wanted to try to build a japanese dictionary but I just wanted to focus on the implementation part. And I was wondering if I could use jiten database files in my project.

**Comment by obfusk** [comment, marked as answer]
Jiten uses data files from JMDict, KanjiDic, Wadoku (for pitch accent), etc.
You can use those as long as you follow the license requirements of those projects, like Jiten does.
There are links in the README: https://github.com/obfusk/jiten#jmdict--kanjidic

**Comment by luckasRanarison** [comment]
Thank you for the quick reply :)! Sorry if I'm bothering you but I have one last question. I'm quite confused with `entry.freq`, it's an INTEGER according to `table_info()` but the values are like string fractions. What are the values based on?

**Reply by obfusk** [reply]
> I'm quite confused with `entry.freq`, it's an INTEGER according to `table_info()` but the values are like string fractions.

Looks like you found a bug. That database field is not used anywhere (all the code uses `entry.rank` instead), so it doesn't actually cause any problems. But thanks!

It's the number of occurrences of that word in the word frequency file(s) divided by the total number of occurrences of all words in the file(s). It is used to calculate the "rank" of the word by sorting the words from highest freq to lowest freq, with the most common word (の iirc) being the one with rank 1. This is used to prioritise the order of words returned when searching.

**Reply by luckasRanarison** [reply]
I see, thank you so much for giving me your time with the detailed explanations! I'm really impressed by the database. I may come again and contribute after learning some python :)
